### PR TITLE
Release 0.2.1

### PR DESCRIPTION
**Pull request by mciantyre**
See #44. I can't build the `0.2.0` release from crates.io. The PR represents the state of the repo when I published a `0.2.1` release.

I started by publishing a `0.2.1-alpha` release. The alpha release was needed to put us in a good state to build and release `0.2.1`. While stepping through the `0.2.1` release, I wrote some notes on how I did the release. I put the notes at `docs/RELEASE.md`.

Show that both crates build when referenced from crates.io:

RAL:
```
cd /tmp
cargo new --lib test-ral
cd test-ral
echo 'imxrt-ral = { "version" = "=0.2.1", features = ["imxrt1062"] }' >> Cargo.toml
cargo build
``` 

HAL:
```
cd /tmp
cargo new --lib test-hal
cd test-hal
echo 'imxrt-hal = { "version" = "=0.2.1", features = ["imxrt1062"] }' >> Cargo.toml
cargo build
```

### CODE REVIEW

**Comment by mciantyre** [comment]
0.2.1 release works in the `teensy4-rs` project. I didn't supply a tag. But, the actual release occurred at 28c995a.

**Comment by mciantyre** [comment]
A change to a markdown file (3a55171d5fe0b55585be67c165f2940bfa8be211) broke the build... 😮

**Comment by mciantyre** [comment]
Nevermind, I tried harder, re-ran the jobs, and they all pass now 😛I'll keep an eye on that.

**Comment by mciantyre** [comment]
I might have introduced CI flakiness by removing the RAL's Cargo.toml file. Here's [the first batch of jobs that failed](https://github.com/mciantyre/imxrt-rs/runs/572554517) on 3a55171. Does the cache not include the auto-generated Cargo.toml file? Let me know what you think; I'm happy to revert a95bf24 if we need.

File: imxrt-hal/Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,12 @@ keywords = ["imxrt", "nxp", "embedded", "no_std"]
 categories = ["embedded", "no-std"]
 license = "MIT/Apache-2.0"
 edition = "2018"
-version = "0.2.0"
+version = "0.2.1"
 
 
 [dependencies]
 cortex-m = { version = "0.6" }
-imxrt-ral = { version = "0.2.0", path = "../imxrt-ral" }
```
Comment (bfrog) [inline imxrt-hal/Cargo.toml:17]: We shouldn't need to remove the path, but we do need to change the version

Comment (bfrog) [reply imxrt-hal/Cargo.toml:17]: See https://github.com/tokio-rs/tokio/blob/master/tokio-tls/Cargo.toml for example

**Pull request merged**